### PR TITLE
Add a contributing guide "CONTRIBUTING.md"

### PR DESCRIPTION
**Pull request by shacharlevy**
Fixes https://github.com/redhat-beyond/HighTeach/issues/5.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+3. Create a working branch and start with your changes!
+
+### Commit your update
+
+Commit the changes once you are happy with them.
+
+### Pull Request
+
+When you're finished with the changes, create a pull request, also known as a PR.
+- Fill the "Ready for review" template so that we can review your PR. This template helps reviewers understand your changes as well as the purpose of your pull request.
```
[the remark this answers is not in the document]
Comment (shacharlevy) [reply CONTRIBUTING.md]: As I found "Ready for review" template is an action for automatically label swapping for draft pull requests.
It's irelevant for our current team workflow. Fixed in the second commit.

**Review by shirazyomtov — APPROVED** [review]
The file looks good to me.
Approved!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+Commit the changes once you are happy with them.
+
+### Pull Request
+
+When you're finished with the changes, create a pull request, also known as a PR.
+- Each Pull Request should focus on single repository principle.
+- <ins>Description-</ins> the PR description should explain what changes have been made, and why. In addition, they should include a link to the relevant `issue`.
+- Each PR should include at most 5 `commits`.
+- Make sure the PR passes CI checks. Run `pipenv run flake8 --max-line-length 120` and `pipenv run python -m pytest -v` inside your VM before submitting a new PR.
+- Each pull request requires the approval of at least <ins>2 team members</ins> before merging.
```
Comment (galitf) [inline CONTRIBUTING.md:55]: I would add that also a mentor needs to review the PR before it can be merged.

Comment (shacharlevy) [reply CONTRIBUTING.md]: Fixed :)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+### Commit your update
+
+Commit the changes once you are happy with them.
+
+### Pull Request
+
+When you're finished with the changes, create a pull request, also known as a PR.
+- Each Pull Request should focus on single repository principle.
+- <ins>Description-</ins> the PR description should explain what changes have been made, and why. In addition, they should include a link to the relevant `issue`.
```
Comment (Omeramsc) [inline CONTRIBUTING.md:52]: same can be said for the commit message

Comment (shacharlevy) [reply CONTRIBUTING.md]: done

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+2. Follow the instructions in the [README.md file](README.md) to set up a working environment.
+
+3. Create a working branch and start with your changes!
+
+### Commit your update
+
+Commit the changes once you are happy with them.
+
+### Pull Request
```
[the remark this answers is not in the document]
Comment (shacharlevy) [reply CONTRIBUTING.md:53]: done

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+When you're finished with the changes, create a pull request, also known as a PR.
+- Each Pull Request should focus on single repository principle.
+- <ins>Description-</ins> the PR description should explain what changes have been made, and why. In addition, they should include a link to the relevant `issue`.
+- Each PR should include at most 5 `commits`.
+- Make sure the PR passes CI checks. Run `pipenv run flake8 --max-line-length 120` and `pipenv run python -m pytest -v` inside your VM before submitting a new PR.
+- Each pull request requires the approval of at least <ins>2 team members</ins> and has to be reviewed by the team mentors before merging.
+
```
[the remark this answers is not in the document]
Comment (shacharlevy) [reply CONTRIBUTING.md:60]: done

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+- Each pull request requires the approval of at least <ins>2 team members</ins> and has to be reviewed by the team mentors before merging. 
+- Examine the PR carefully, and with respect to what it is supposed to achieve.
+- If you want to request changes to specific lines in a file please use the "Start a review" button.
+- If you want to request changes, mark your review as "Request changes" - please do not use a generic comment.
+- Use polite, respectful and clear language when commenting on or reviewing a PR.
+- If your PR recieved a request for changes or a comment, please notify the reviewer about your actions
+   (what changes you made, or if you don't think they're necessary - why you think so).
+- Changes made as part of the review process (i.e. fixes to the content of the original commit),
+  should be added to the original commit (e.g. using `commit --amend`), rather than creating a new commit.
```
[the remark this answers is not in the document]
Comment (shacharlevy) [reply CONTRIBUTING.md]: Ok, I removed this sentence from the PR code review section.

**Review by Yardengz — APPROVED** [review]
lgtm

**Review by natifridman — APPROVED** [review]

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+In addition, they should include a link to the relevant `issue`.
+
+Please use the guide below in order to follow our commit message convention:
+* [commit convention](https://cbea.ms/git-commit/#seven-rules)
+
+### Pull Request
+
+When you're finished with the changes, create a pull request, also known as a PR.
+
+- Each Pull Request should focus on single repository principle.
```
Comment (Omeramsc) [inline CONTRIBUTING.md:57]: What does "repository principle" means?

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+- Each PR should include at most 5 `commits`.
+- Make sure the PR passes CI checks. Run `pipenv run flake8 --max-line-length 120` and `pipenv run python -m pytest -v` inside your VM before submitting a new PR.
+
+PR documentation:
+
+- Every PR should have a meaningful and concise title that clearly describes its purpose.
+- Every PR should have a meaningful and concise description that provides relevant information on the PR.
+- Every PR should be linked to an issue (or issues), to make its purpose clear and increase its visibility (see [Issues](#issues)).
+- Assign other teammates as well as mentors as reviewers on your PR.
```
Comment (Omeramsc) [inline CONTRIBUTING.md:66]: maybe rephrase teammates to contributors and mentors to maintainers, as this is a public open source project.

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+
+PR code review:
+
+- Each pull request requires the approval of at least <ins>2 team members</ins> and has to be reviewed by the team mentors before merging.
```
Comment (Omeramsc) [inline CONTRIBUTING.md:70]: rephrase here as well

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+- Every PR should have a meaningful and concise description that provides relevant information on the PR.
+- Every PR should be linked to an issue (or issues), to make its purpose clear and increase its visibility (see [Issues](#issues)).
+- Assign other teammates as well as mentors as reviewers on your PR.
+
+PR code review:
+
+- Each pull request requires the approval of at least <ins>2 team members</ins> and has to be reviewed by the team mentors before merging. 
+- Examine the PR carefully, and with respect to what it is supposed to achieve.
+- If you want to request changes to specific lines in a file please use the "Start a review" button.
+- If you want to request changes, mark your review as "Request changes" - please do not use a generic comment.
```
Comment (Omeramsc) [inline CONTRIBUTING.md:73]: oopsie :P

Comment (Omeramsc) [reply CONTRIBUTING.md:73]: but you're right, will post comments as review starting next review

**Review by Omeramsc — CHANGES_REQUESTED** [review]
Overall looking good, left some minor comment. let's get it merged today!

File: CONTRIBUTING.md (hunk 7 above)
Comment (shacharlevy) [reply CONTRIBUTING.md]: It means that we should apply the principle of single responsibility. If our code is doing more than one thing we should break it into different PRs.
I reworded it and squashed it to the previous commit that  contains all the minors updates of the CONTRIBUTING.md file.

File: CONTRIBUTING.md (hunk 8 above)
Comment (shacharlevy) [reply CONTRIBUTING.md]: Fixed :)

File: CONTRIBUTING.md (hunk 9 above)
Comment (shacharlevy) [reply CONTRIBUTING.md]: Fixed :)

**Comment by natifridman** [comment]
Please squash the commits so we can merge it
